src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
+     * Get the label for a CustomField.
+     *
+     * To be called in getColumnNameForAssociation() for instances using
+     * CubeTools\CubeCustomFieldsBundle\Entity\EntityCustomField
+     *
+     * @param AuditLog $currentVersion
+     *
+     * @return string
+     */
+    protected function getCustomFieldLabel(AuditLog $currentVersion)
+    {
+        if (!$currentVersion->getTarget() || 'custom_fields' !== $currentVersion->getTarget()->getTbl()) {
+            return;
+        }
+        $cfId = $currentVersion->getTarget()->getFk();
+        if (isset($this->instanceCache['customfield_labels'][$cfId])) {
+            return $this->instanceCache['customfield_labels'][$cfId];
+        }
+
+        $cfQb = $this->em->getRepository(AuditLog::class)->createQueryBuilder('c')
+            ->join('c.source', 's')
+            ->where("c.action = 'insert'")
+            ->andWhere('s.fk = :cfId')->setParameter('cfId', $cfId)
+            ->andWhere("s.tbl = 'custom_fields'")
+        ;
+        $cfLogs = $cfQb->getQuery()->getResult();
+        $fieldId = null;
+        foreach ($cfLogs as $log) {
+            if (isset($log->getDiff()['fieldId'])) {
+                $fieldId = $log->getDiff()['fieldId']['new'];
+                break;
+            }
+        }
+
+        if (is_null($fieldId)) { // fallback
+            $fieldId = 'customfield_'.$cfId;
+        }
+        $this->instanceCache['customfield_labels'][$cfId] = $fieldId;
+
+        return $fieldId;
+    }
+
+    /**
+     * Filters the diff format of CustomFields.
+     *
+     * To be used in overwritten getAllVersionsDiff() after calling it from parent class.
+     * Rewrites the values as simple value instead of array.
+     *
+     * @param array $diff
+     *
+     * @return array modified $diff
+     */
+    protected function filterCustomFieldsAllVersionsDiff($diff)
+    {
+        if (empty($this->instanceCache['customfield_labels'])) {
+            return $diff;
+        }
+        $cfLabels = array_unique($this->instanceCache['customfield_labels']);
+        foreach ($diff as $key => $changeEntry) {
+            $changes = $changeEntry['changes'];
+            foreach ($cfLabels as $label) {
+                if (!isset($changes[$label])) {
+                    continue;
+                }
+                $value = isset($changes[$label][self::KEY_ADD]) ? $changes[$label][self::KEY_ADD] : '';
+                $diff[$key]['changes'][$label] = $value;
+            }
+        }
+
+        return $diff;
+    }
+
+    /**
+     * Returns doctrines metadata for the main entity.
+     *
+     * @return \Doctrine\ORM\Mapping\ClassMetadata
+     */
+    private function getEntitiesClassMetaData()
+    {
+        return $this->em->getClassMetadata($this->cache['class']);
+    }
+
+    /**
+     * Caches a value as current.
+     *
+     * @param Association $ass
+     * @param string      $value
+     */
+    private function setCachedAssociationValue(Association $ass, $value)
```

Comment:
Hint (for this and getCachedAssociationValue): why do not use protected function? It would ease our work if we would like somewhere in the future extend this class

Reply:
because I look the cachedAssociationValue as private, nobody should handle with it. But the instanceCache is protected, so this is not really consistent.

Reply:
I prefer always to use protected, unless there is a good reason to make private one. We can also discuss it on Cube Team Meeting.